openapi/definitions.yaml

```diff
@@ -544,6 +546,139 @@ InternalMessageResponseWithContent:
   required:
     - message
 
+RCConfigurationResponse:
+  allOf:
+    - $ref: "#/RCConfigurationBase"
+    - $ref: "#/RCConfigurationEnvironment"
+  
+RCConfigurationBase:
+  type: object
+  properties:
+    userId: 
+      type: string
+      description: User id.
+    configurationId:
+      type: string
+      description: Ulid string.
+      format: Ulid
+      x-import: '@pagopa/ts-commons/lib/strings'
+      example: 01ARZ3NDEKTSV4RRFFQ69G5FAV
+    name:
+      type: string
+      description: Configuration name.
+    description:
+      type: string
+      description: Configuration description.
+    hasPrecondition:
+       $ref: "#/HasPrecondition"
+    disableLollipopFor:
+      type: array
+      items:
+        type: string
+        description: User for which lollipop shoulg be disabled
+    isLollipopEnabled:
+      type: boolean
+      description: Flag to check if lollipop is enabled
+  required:
+    - userId
+    - configurationId
+    - name
+    - description
+    - hasPrecondition
+    - disableLollipopFor
+    - isLollipopEnabled
+
+RCConfigurationEnvironment:
+  x-one-of: true
+  allOf:
+    - $ref: "#/RCConfigurationWithTestEnvironment"
+    - $ref: "#/RCConfigurationWithProdEnvironment"
+    - $ref: "#/RCConfigurationWithProdAndTestEnvironment"
+
+RCConfigurationWithTestEnvironment:
+  type: object
+  properties:
+    testEnvironment:
+      $ref: "#/RCConfigurationTestEnvironment"
+  required:
+    - testEnvironment
+
+RCConfigurationWithProdEnvironment:
+  type: object
+  properties:
+    prodEnvironment:
+      $ref: "#/RCConfigurationProdEnvironment"
+  required:
+    - prodEnvironment
+
+RCConfigurationWithProdAndTestEnvironment:
+  type: object
+  properties:
+    prodEnvironment:
+      $ref: "#/RCConfigurationProdEnvironment"
+    testEnvironment:
+      $ref: "#/RCConfigurationTestEnvironment"
+  required:
+    - prodEnvironment
+    - testEnvironment
+
+RCConfigurationProdEnvironment:
+  type: object
+  properties:
+    baseUrl:
+      type: string
+      description: Base url of the server.
+    detailsAuthentication:
+      $ref: "#/RCAuthenticationConfig"
+  required:
+    - baseUrl
+
+RCAuthenticationConfig:
+  type: object
+  properties:
+    headerKeyName:
+      type: string
+      description: header key name
+    key:
+      type: string
+      description: authentication config key
+    type:
+      type: string
+      description: authentication config type
```

Comment:
Take care, those properties should be `NonEmptyString`, we need to add `minLength: 1`